sklearn/cross_decomposition/_pls.py

```diff
@@ -532,7 +532,7 @@ class PLSRegression(_PLS):
     ----------
     n_components : int, default=2
         Number of components to keep. Should be in `[1, min(n_samples,
-        n_features, n_targets)]`.
+        n_features)]`.
```

Comment:
From looking at the code, regression is bounded only by `n_features`:

https://github.com/scikit-learn/scikit-learn/blob/463d1665e628628e837d11b323d86cadaae27081/sklearn/cross_decomposition/_pls.py#L241-L250

In that case, I think bound is `[1, n_features]` for regression.

Reply:
You're right, I confirmed in the cited technical report that the rank of X.T X is the upper limit. I've updated the docstring and the user guide for PLSRegression to reflect that.